core/src/main/java/tc/oc/pgm/core/CoreModule.java

```diff
@@ -71,6 +71,20 @@ public MatchModule createMatchModule(Match match) {
       return ImmutableList.of(RegionModule.class, TeamModule.class);
     }
 
+    public ImmutableSet<SelfIdentifyingFeatureDefinition> parseModeSet(
+        Node node, MapFactory context) throws InvalidXMLException {
+      ImmutableSet.Builder<SelfIdentifyingFeatureDefinition> modes = ImmutableSet.builder();
+      for (String modeId : node.getValue().split("\\s")) {
+        SelfIdentifyingFeatureDefinition mode =
+            context.getFeatures().get(modeId, SelfIdentifyingFeatureDefinition.class);
+        if (mode == null) {
+          throw new InvalidXMLException("No mode with ID '" + modeId + "'", node);
+        }
+        modes.add(mode);
+      }
+      return modes.build();
```

Comment:
```suggestion
      ImmutableSet.Builder<Mode> modes = ImmutableSet.builder();
      for (String modeId : node.getValue().split("\\s")) {
        Mode mode = context.getFeatures().get(modeId, Mode.class);
        if (mode == null) {
          throw new InvalidXMLException("No mode with ID '" + modeId + "'", node);
        }
        modes.add(mode);
      }
      return modes.build();
```
Get modes, not `SelfIdentifyingFeatureDefinition`s